sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -0,0 +1,574 @@
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private static void WriteGetFunctionsMetadataAsyncMethod(IndentedTextWriter indentedTextWriter, SyntaxReceiver receiver, Compilation compilation)
+        {
+            indentedTextWriter.WriteLine("public Task<ImmutableArray<IFunctionMetadata>> GetFunctionMetadataAsync(string directory)");
+            indentedTextWriter.WriteLine("{");
+            indentedTextWriter.Indent++;
+
+            // create list of IFunctionMetadata and populate it
+            indentedTextWriter.WriteLine("var metadataList = new List<IFunctionMetadata>();");
+            AddFunctionMetadataInfo(indentedTextWriter, receiver, compilation);
+            indentedTextWriter.WriteLine("return Task.FromResult(metadataList.ToImmutableArray());");
+
+            indentedTextWriter.Indent--;
+            indentedTextWriter.WriteLine("}");
+        }
+
+        private static void AddFunctionMetadataInfo(IndentedTextWriter indentedTextWriter, SyntaxReceiver receiver, Compilation compilation)
+        {
+            var assemblyName = compilation.Assembly.Name;
+            var scriptFile = Path.Combine(assemblyName + ".dll");
+
+            // Loop through the candidate methods (these are mthods suspected to be Functions, stored in the SyntaxReceiver)
+            foreach (MethodDeclarationSyntax method in receiver.CandidateMethods)
+            {
+                var functionClass = (ClassDeclarationSyntax)method.Parent!;
+                var functionName = functionClass.Identifier.ValueText;
+                var entryPoint = assemblyName + "." + functionName + "." + method.Identifier.ValueText;
+                var bindingsListName = functionName + "RawBindings";
+
+                // collect Bindings
+                indentedTextWriter.WriteLine($"var {bindingsListName} = new List<string>();");
+                AddBindingInfo(indentedTextWriter, method, compilation, functionName);
+                indentedTextWriter.WriteLine($"var {functionName} = new DefaultFunctionMetadata(Guid.NewGuid().ToString(), \"dotnet-isolated\", \"{functionName}\", \"{entryPoint}\", {functionName}RawBindings, \"{scriptFile}\");");
+                indentedTextWriter.WriteLine($"metadataList.Add({functionName});");
+            }
+        }
+
+        private static void AddBindingInfo(IndentedTextWriter indentedTextWriter, MethodDeclarationSyntax method, Compilation compilation, string functionName)
+        {
+            var model = compilation.GetSemanticModel(method.SyntaxTree);
+            var bindingCount = 0;
+            var bindingsListName = functionName + "RawBindings";
+
+            // Loop through the parameters in this Function to find the bindings
+            foreach (ParameterSyntax parameter in method.ParameterList.Parameters)
```

Comment:
We probably also need to loop through `method.AttributeLists` collection to generate output binding data.

    [Function("Function2")]
    [QueueOutput("myqueue", Connection ="M")]
    public void Run([QueueTrigger("myqueue", Connection = "M")] string m)
    {
         // 
    }

The `QueueOutput` output binding should produce binding entry. 
Consider adding a test for case like that.